### PR TITLE
React Caption Project

### PR DESCRIPTION
**Pull request by ctokoli**
These are the features implemented in this caption project. 

- The project is a single-page application (SPA) built with React and Redux and Covid 19 API.
- The data retrieved from the Covid-19 API was stored in the Redux store.
-  The data retrieved from the API was filtered.
- Every page (the main page and the pages for each item) have a unique route within the SPA.
- The project was deployed and it accessible online using [Render](https://render.com/).
- Added a link of the online version of the app to the README file.
- Created a unit tests for pure functions in the Redux code.
- Created integration tests for the application using the React Testing Library

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,172 @@
...
-The build is minified and the filenames include the hashes.\
-Your app is ready to be deployed!
+<!-- LIVE DEMO -->
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+## 🚀 Live Demo <a name="live-demo"></a>
 
-### `npm run eject`
+- [Live Demo Link](https://covid-stats-p1cd.onrender.com/)
+<!-- - N/A -->
```
Comment (rotimiazeez) [inline README.md:53]: Please ensure you add a link to a short video presentation of your project (not longer than 5mins) in your README file. Kindly ensure it contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear and contains a demo of the project, highlighting all the project features.

File: src/App.css [hunk 2]
```diff
@@ -1,3 +1,138 @@
...
+  .top-btns {
+    padding: 4px;
+    padding-left: 9px;
+    text-align: center;
+  }
+
+  button.btn.btn-outline-success {
+    margin-bottom: 5px;
+  }
 }
```
Comment (rotimiazeez) [inline src/App.css]: Please ensure your design looks more like the original design from Behance, for a better UI.

Your Design  |  Behance Design
Screenshot from 2023-08-12 19-06-19  |  Screenshot from 2023-08-12 19-08-23

File: src/tests/__snapshots__/Components.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    >
+      <h3 />
+    </div>
+  </div>
+  <div
+    className="inactive-data"
+  >
+    <h5>
+      Total Population
+    </h5>
```
Comment (rotimiazeez) [inline src/tests/__snapshots__/Components.test.js.snap]: Kindly ensure all your app test cases are passing for a more professional codebase.
Screenshot from 2023-08-12 19-05-26

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @ctokoli 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good functioning app 👌🏽
- Good code structure 👏🏾
- Good use of Redux for state management 🙂
- Good application of filter stateless components for filtering data received from the API 🤝🏽
- Good job acknowledging the owner of the original design.
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: README.md [hunk 4]
```diff
@@ -1,70 +1,177 @@
...
 
-**Note: this is a one-way operation. Once you `eject`, you can't go back!**
+## 🚀 Video Recording <a name="video"></a>
 
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+- [Video ](https://drive.google.com/file/d/1gJdc3TPpkgGaqFrCs9_UFWwKBBZWQsVk/view?usp=sharing)
```
Comment (Lizdev-05) [inline README.md]: - [ ] As per the request of the previous reviewer, please create a brief video presentation of your project that is no longer than 5 minutes in duration. Your current presentation exceeds this time limit, so kindly record a new video that accurately showcases your project within the given time constraints. :relaxed:

File: src/App.css [hunk 5]
```diff
@@ -1,3 +1,151 @@
...
+.deaths {
+  background-color: #fd5294;
+}
+
+a img {
+  width: 27px;
+}
+
+section.top-page {
+  display: flex;
```
Comment (Lizdev-05) [inline src/App.css]: - [ ] The previous code reviewer has asked you to ensure that your app's home page matches the design that was given. However, upon comparing the two designs, it appears that the background colors of the countries' list should differ from each other, just like in the given design.

### Your design
image

### Given design
image

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @ctokoli , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Good Presentation Video ✔️ 
- The student deploys the application using Render✔️
- The project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly✔️
- Followed [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-\(Free-PSDs\))✔️
- No Linter errors ✔️
-  App has unit tests for several components of the app and some pure functions in Redux code.✔️
- App has integration tests using React Testing Library ✔️


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**